### ISSUE TITLE
Pango Markup Not Correctly Rendered in Tooltip

### ISSUE THREAD

**Issue by Fr4nk1inCs** [Alexays/Waybar#3375]
HTML/Pango Markup in tooltip is not rendered correctly in waybar 0.10.2.

For example, I have a spotify module that prints this to output:
```json
{
  "text": "{icon} {artist} - {title}",
  "tooltip": "<b><u>Spotify</u></b>\n{title}\n{artist} - {album}",
  "class": "custom-spotify",
  "alt": "spotify"
}
```

This is how the tooltip is rendered in v0.9.24 (**<u>Spotify</u>** ...):
![rect3](https://github.com/Alexays/Waybar/assets/74861140/94e09557-d41c-4b0e-9a2d-2560675d2ca4)

However, these markup tags are ignored and print out in v0.10.2 (\<b\>\<u\>Spotify\<\\u\>\<\\b\> ...):
![rect1](https://github.com/Alexays/Waybar/assets/74861140/eaf3817b-6227-4f4d-86cb-465cd058171f)

~~And I think this may be related to issues that `calendar` tooltip of the `clock` module is not correctly aligned when using CJK locale (e.g. https://github.com/Alexays/Waybar/issues/3132 and https://github.com/Alexays/Waybar/issues/1321#issuecomment-2027596828) since it works fine in v0.9.24.~~(just checked, not related)

**Comment by Fr4nk1inCs** [comment]
I have ran a simple `git bisect` and found that commit https://github.com/Alexays/Waybar/commit/2673a5a4f1e2cee9f2edee57ca278697fdfdaf50 might be the reason.

**Comment by RobertMueller2** [comment]
Have you checked whether this is still happening with 0.10.3? I'm thinking a453ea3c70195803e9962a97f9243c0c78d4ecdc might have fixed it, which is included in 0.10.3.

**Comment by Fr4nk1inCs** [comment]
Nope, still `<b><u>Spotify<\u><\b>`, tried both v0.10.3 (from Arch Linux Packages) and the latest commit (https://github.com/Alexays/Waybar/commit/99971556178ae504b7ba7ab559952f781925f7da)

**Comment by RobertMueller2** [comment]
OK, it works for me. I'm using a limited repro.

![satty-20240701-17:14:07](https://github.com/Alexays/Waybar/assets/29629287/799c62b6-bc41-4035-9f43-467fc89c9970)



```
{
    "layer": "top", // Waybar at top layer
    "position": "top", // Waybar position (top|bottom|left|right)
    "height" : 50,
    "modules-center": ["custom/test-output"],

    "custom/test-output": {
      "format": "{}",
      "exec": "/home/user/tmp/waybar-test-config/test-output3.sh",
      "return-type":"json",
      "interval": 1,
    },
}
```

```

window {
    background-color: transparent;
}

#custom-test-output {
    color: #ffffff;
    background-color: #ff0000;
}
```

```
#!/bin/sh

TEXT="  Moguai - Freaks"
TOOLTIP="<b><u>Spotify</u></b>\nMoguai\nFreaks"
CLASS="custom-spotify"
ALT="spotify"

printf '{"text": "%s", "tooltip": "%s", "class" : "%s", "alt" : "%s"}\n' "$TEXT" "$TOOLTIP" "$CLASS" "$ALT"
```

Could you kindly share your config?

**Comment by Fr4nk1inCs** [comment]
Found a small problem in my config, I accidentally add `"escape": true` in the module.

Thx for you time.

**Comment by RobertMueller2** [comment]
No problem, that explains it :) good to hear it's working now.

**Comment by Fr4nk1inCs** [comment]
Sorry to reopen this, but I just encounter the same issue when I migrate this custom spotify widget into `mpris` one.


```json
"mpris": {
  "interval": 1,
  "format": "{status_icon} {artist} - {title}",
  "format-stopped": "",
  "tooltip-format": "<b>{player}</b>\n{title} {position}/{length}\n{artist} - {album}",
  "tooltip-format-stopped": "",
  "status-icons": {
    "playing": "󰏤",
    "paused": "󰐊",
    "stopped": "",
  },
},
```

This happens on both waybar v0.10.3 (from Arch Linus Packages) and the latest commit https://github.com/Alexays/Waybar/commit/15e1547661bfc5fe9b3d45bb0d9cea11cf07db7f .

@RobertMueller2 Can you reproduce this?

**Comment by RobertMueller2** [comment]
Yes. Mpris is using `set_tooltip_text` rather than `set_tooltip_markup`.

https://github.com/Alexays/Waybar/blob/15e1547661bfc5fe9b3d45bb0d9cea11cf07db7f/src/modules/mpris/mpris.cpp#L726

Somebody would have to change that. That doesn't look difficult for the mpris module. And then the next issue is about the taskbar module, or the battery module, or.... ;) 

The challenge here is to do it as centrally as possible, perhaps in AModule, and then achieve a sensible result for all 44 places in the modules that set a tooltip. They might even have different config options for formatting, which could break existing configs. The tooltip content generation has to remain within the modules, because that's very specific, but whether markup is used or text could be in AModule based on a config option.

**Comment by Cprakhar** [comment]
@RobertMueller2 Any updates?

**Comment by RobertMueller2** [comment]
> @RobertMueller2 Any updates?

As far as I remember I provided a few module specific fixes in PR #3460 2 years ago which I rebased several times until half a year ago, which means it was ready to merge for one and a half years.

I can't comment on why it wasn't, you'd have to ask a maintainer about that.

I can rebase it again IF it has a chance of being merged. EDIT: And when Waybar master branch actually builds. ;)

EDIT2: looks like the PR is still mergeable, so I don't see anything for me to do.

**Comment by Alexays** [comment]
Closing — this is fixed/implemented on current master and will be in the next release. Please reopen if it still occurs after updating. 🙏
